src/services/EventService.ts

```diff
@@ -2,12 +2,43 @@ import { Event, AppUser, Attendance, RSVP } from '@Entities';
 import { AttendanceService, AttendanceServiceImpl } from './AttendanceService';
 import { RSVPService, RSVPServiceImpl } from './RSVPService';
 
-import { getRepository } from 'typeorm';
-import { MultipleAttendanceQuery } from '@Payloads';
+import { getRepository, FindManyOptions, Not } from 'typeorm';
+import { MultipleAttendanceQuery, MultipleEventQuery } from '@Payloads';
 
 export class EventService {
   constructor(private attendanceService: AttendanceService, private rsvpService: RSVPService) {}
 
+  private buildMultipleEventQuery(
+    multipleEventQuery: MultipleEventQuery,
+    canShowPending: boolean
+  ): FindManyOptions<Event> {
+    const { pending, ready, complete } = multipleEventQuery;
+    const query: FindManyOptions<Event> = {};
+    const whereArr = [];
+
+    if (canShowPending) {
+      if (pending) {
+        whereArr.push({ status: 'pending' });
+      }
+    } else {
+      if ((ready && complete) || (!ready && !complete)) {
+        whereArr.push({ status: Not('pending') });
```

Comment:
whats this

Reply:
This is because Not('pending') queries include 'ready' and 'complete' statuses (so it gets anything that is not 'pending'). So let's say a user is not permitted to see pending events, but say they only want to see events that are ready but not events that are complete. Then, without that check, Not('pending') will erroneously return both events that are ready and events that are complete, even though we only wanted events that are ready. By having that check, we are making sure that Not('pending') = 'ready' + 'complete' query is made only when we have both ready and complete = true in query params or when we don't have either of them, since no query params = show every event but for some users we don't want them to see pending so they'll only be seeing ready and complete with this.